src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -340,21 +340,47 @@ trait ContextErrors {
 
       //typedSelect
       def NotAMemberError(sel: Tree, qual: Tree, name: Name) = {
-        def errMsg = {
+        import util.{ EditDistance, StringUtil }
+        def errMsg: String = {
           val owner            = qual.tpe.typeSymbol
           val target           = qual.tpe.widen
           def targetKindString = if (owner.isTypeParameterOrSkolem) "type parameter " else ""
           def nameString       = decodeWithKind(name, owner)
           /* Illuminating some common situations and errors a bit further. */
           def addendum         = {
+            val companionSymbol: Symbol = {
+              if (name.isTermName && owner.isPackageClass)
+                target.member(name.toTypeName)
+              else NoSymbol
+            }
             val companion = {
-              if (name.isTermName && owner.isPackageClass) {
-                target.member(name.toTypeName) match {
-                  case NoSymbol => ""
-                  case sym      => "\nNote: %s exists, but it has no companion object.".format(sym)
-                }
+              if (companionSymbol == NoSymbol) ""
+              else s"\nnote: $companionSymbol exists, but it has no companion object."
+            }
+            // find out all the names available under target within 2 edit distances
+            lazy val alternatives: List[String] = {
+              val editThreshold = 2
+              val x = name.toString
+              if (x.size < 2) Nil
+              else {
+                target.nonPrivateMembers.iterator
+                  .filterNot(m => m.isConstructor)
+                  .map(_.name.decode)
+                  .filter(n => (n.length > 2) &&
+                    (math.abs(n.length - x.length) <= editThreshold) &&
+                    (n != x) &&
+                    !n.contains("$") &&
+                    EditDistance.levenshtein(n, x) <= editThreshold)
+                  .distinct.toList
```

Comment:
We should probably only suggest term/type names depending on whether `name` is a term/type name.